streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ProjectServiceImpl.java

```diff
@@ -456,37 +276,26 @@ private void eachFile(File file, List<Map<String, Object>> list, Boolean isRoot)
         }
     }
 
-    private boolean projectBuild(Project project, String socketId) {
-        StringBuilder builder = tailBuffer.get(project.getId());
-        int code = CommandUtils.execute(project.getMavenWorkHome(),
-            Collections.singletonList(project.getMavenArgs()),
-            (line) -> {
-                builder.append(line).append("\n");
-                if (tailOutMap.containsKey(project.getId())) {
-                    if (tailBeginning.containsKey(project.getId())) {
-                        tailBeginning.remove(project.getId());
-                        Arrays.stream(builder.toString().split("\n"))
-                            .forEach(out -> WebSocketEndpoint.writeMessage(socketId, out));
-                    }
-                    WebSocketEndpoint.writeMessage(socketId, line);
-                }
-            });
-        closeBuildLog(project.getId());
-        log.info(builder.toString());
-        tailBuffer.remove(project.getId());
-        return code == 0;
-    }
-
     @Override
-    public void tailBuildLog(Long id) {
-        this.tailOutMap.put(id, Byte.valueOf("0"));
-        this.tailBeginning.put(id, Byte.valueOf("0"));
+    public String getBuildLog(Long id) {
+        Path logPath = Paths.get(getBuildLogPath(id));
+        if (!Files.exists(logPath)) {
+            String errorMsg = String.format("Build log file(fileName=%s) not found, please build first.", logPath);
+            log.info(errorMsg);
+            return errorMsg;
+        }
+        byte[] fileContent;
+        try {
+            fileContent = Files.readAllBytes(logPath);
+        } catch (IOException e) {
+            log.error("Read build log file(fileName={}) caused an exception: ", logPath, e);
+            return Strings.EMPTY;
+        }
+        return new String(fileContent, StandardCharsets.UTF_8);
```

Comment:
In this way, it is not recommended to get the log contents by reading the complete contents of the file. If the file is large, it may have a negative impact on the streampark application.